native/imgproc.cc

```diff
@@ -92,4 +92,11 @@ void cv_compare_hist(cv::Mat* first_image, cv::Mat* second_image, int method, Re
     *result = Result<double>::FromFunction(
         [first_image, second_image, method]() { return cv::compareHist(*first_image, *second_image, method); });
 }
+
+void cv_canny(cv::Mat* image, cv::Mat* edges, double threshold1, double threshold2, int aperture_size, bool l2_gradient, Result<cv::Mat*>* result) {
+    *result = Result<cv::Mat*>::FromFunction([image, edges, threshold1, threshold2, aperture_size, l2_gradient]() { 
```

Comment:
You return `edges` as output parameter and as function result. I don't think it's a good thing, so I propose to add CEmptyResult which works the same way as `CResult<T>` but that doesn't have any result (I think, it's better than having CResult<Some_ignored_type_parameter>). E.g. something like:

```cpp
struct EmptyResult {
    CDisposableString error;

    static EmptyResult FromFunction(std::function<void()> function) {
        char* error = nullptr;
        try {
            function();
        } catch (cv::Exception& e) {
            const char* err_msg = e.what();
            auto len = std::strlen(err_msg);
            error = new char[len + 1];
            std::strcpy(error, err_msg);
        }
        return Result{CDisposableString{error}};
    }
};
```

So signature of the function could be `EmptyResult cv_canny(...)`